### ISSUE TITLE
Deploy command fails with "TypeError: memoizeOne is not a function"

### ISSUE THREAD

**Issue by Nils-Kolvenbach** [netlify/cli#2210]
**Describe the bug**

We use Netlify CLI inside GitLab pipelines to deploy static files to Netlify. All of them are currently (netlify-cli@3.21.6) failing with the error "TypeError: memoizeOne is not a function". The last successful deployment was 17 hours ago with netlify-cli@3.21.4. Since then, nothing on our end changed.

**To Reproduce**

Reproducing it could be challenging. This only happens inside our pipelines. When I try to deploy from my local machine (with netlify-cli@3.21.6) it works as expected. Please let me know what you would need to debug this issue.

**Configuration**

```
yarn run v1.22.5
$ npx envinfo --system --binaries --npmPackages netlify-cli --npmGlobalPackages netlify-cli
  System:
    OS: Linux 3.10 Debian GNU/Linux 9 (stretch) 9 (stretch)
    CPU: (4) x64 Intel(R) Xeon(R) CPU E5-2620 v2 @ 2.10GHz
    Memory: 2.29 GB / 15.51 GB
    Container: Yes
    Shell: 4.4.12 - /bin/bash
  Binaries:
    Node: 15.14.0 - /tmp/yarn--1619171047938-0.7351852524007791/node
    Yarn: 1.22.5 - /tmp/yarn--1619171047938-0.7351852524007791/yarn
    npm: 7.7.6 - /usr/local/bin/npm
Done in 2.06s.
```

**Expected behavior**

Netlify CLI should be able to deploy without failing.

**CLI Output**

```
$ yarn global add netlify-cli
yarn global v1.22.5
[1/4] Resolving packages...
warning netlify-cli > @netlify/plugin-edge-handlers > rollup-plugin-node-polyfills > rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
warning netlify-cli > copy-template-dir > readdirp > micromatch > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
warning netlify-cli > copy-template-dir > readdirp > micromatch > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
[2/4] Fetching packages...
info @netlify/traffic-mesh-agent-darwin-x64@0.27.10: The platform "linux" is incompatible with this module.
info "@netlify/traffic-mesh-agent-darwin-x64@0.27.10" is an optional dependency and failed compatibility check. Excluding it from installation.
info @netlify/traffic-mesh-agent-win32-x64@0.27.10: The platform "linux" is incompatible with this module.
info "@netlify/traffic-mesh-agent-win32-x64@0.27.10" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "netlify-cli > @octokit/rest > @octokit/plugin-request-log@1.0.3" has unmet peer dependency "@octokit/core@>=3".
[4/4] Building fresh packages...
success Installed "netlify-cli@3.21.6" with binaries:
      - ntl
      - netlify
Done in 48.05s.
$ netlify deploy --dir "..." --site "..." --auth "..." --prod
    TypeError: memoizeOne is not a function
ERROR: Job failed: exit code 1

```
**Additional context**

**Comment by Nils-Kolvenbach** [comment]
https://github.com/netlify/cli/pull/2204 and https://github.com/netlify/cli/issues/2202 seem to be related. Sorry for not commenting before mentioned issue instead of opening a new one. I just searched for open issues

**Comment by JGAntunes** [comment]
Hey @Nils-Kolvenbach 👋 thanks for raising this. This is most likely related with that same problem in `@netlify/build` - https://github.com/netlify/build/issues/2642 & https://github.com/netlify/build/pull/2643 - we're working on fixing it and will update this thread once we cut a new `cli` release 👍

**Comment by JGAntunes** [comment]
This should be addressed in our latest release 3.21.7 - https://www.npmjs.com/package/netlify-cli/v/3.21.7 - via https://github.com/netlify/cli/pull/2207 👍 Let us know if you still have any issues @Nils-Kolvenbach
